operational_analysis/methods/plant_analysis.py

```diff
@@ -774,15 +775,22 @@ def run_regression(self, n):
             self._mse_score[n] = mean_squared_error(reg_data[:,-1], predicted_y)
             return reg
         # Machine learning models
-        else: 
+        else:
             ml = MachineLearningSetup(self.reg_model, **self.ml_setup_kwargs)
-            ml.hyper_optimize(np.array(reg_data[:,0:-1]), reg_data[:,-1], n_iter_search = 5, report = False, cv = KFold(n_splits = 2))
-            predicted_y = ml.random_search.predict(np.array(reg_data[:,0:-1]))
+            # Memoized approach for optimized hyperparameters
+            if self._run.reanalysis_product in self.opt_model:
+                self.opt_model[(self._run.reanalysis_product)].fit(np.array(reg_data[:,0:-1]), reg_data[:,-1])
```

Comment:
Would be useful to add a comment saying the last column of reg_data is the target variable for regression. This was a little confusing to me re-reading the code.

Reply:
I've now added a comment!